### PR TITLE
DBTP-387 Migrate to DBT PaaS

### PR DESCRIPTION
**Pull request by WillGibson**
**N.B. Merging should wait until we are ready to flip over to running in DBT PaaS for real.**

Basically what we have come to expect in a migration now, plus breaking up `settings.py` into three files with some inheritance to avoid loads of duplication for the build one.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1 +1,11 @@
-# url-protection-checker
\ No newline at end of file
+# URL Protection Checker
+
+Maintains a list of URLs that should be protected by the IP Filter and alerts if it ever finds someone has been able to access them from an IP address which has not been whitelisted.
```
Comment (ejayesh) [inline README.md:3]: This should be:
Maintains a list of URLs that are publicaly accessible on Gov PaaS and alerts if its not protecteed by Staff SSO or the IP Filter, unless the URL is whitelisted in this app

**Review by ejayesh — DISMISSED** [review]
small change

File: README.md (hunk 1 above)
Comment (WillGibson) [reply README.md]: Slightly reworded, from your suggestion, but done.

**Comment by WillGibson** [comment]
Starting a new pull request to merge feature/DBTP-387-migrate-to-dbt-paas into develop.

File: config/sample_env
Comment (WillGibson) [inline config/sample_env:1]: We have `.env.sample` now.

File: core/admin_views.py [hunk 2]
```diff
@@ -24,7 +24,7 @@ def admin_login_view(request):
 
     if request.user.is_authenticated:
         if not request.user.is_staff:
-            raise PermissionDenied
+            raise PermissionDenied('User is not staff')
```
Comment (WillGibson) [inline core/admin_views.py:27]: This simple addition would have saved me some time.

File: user/migrations/0003_alter_user_first_name.py [hunk 3]
```diff
@@ -0,0 +1,18 @@
+# Generated by Django 3.2.18 on 2023-09-14 15:48
```
Comment (WillGibson) [inline user/migrations/0003_alter_user_first_name.py:1]: There were changes to the model that was not reflected in the database.

File: config/settings/build.py [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+        "username": "does-not-matter",
+        "password": "does-not-matter",
+        "dbname": "does-not-matter",
+        "engine": "postgres",
+        "port": 5432,
+        "host": "localhost"
+    }
+}
+
+AUTHBROKER_URL = "dont-use-in-prod"
```
Comment (antroy-madetech) [inline config/settings/build.py:18]: What's the purpose of all of these settings if they are not actually used?

**Comment by WillGibson** [comment]
Closing this as very stale and I expect the migration team will start from scratch anyway.

**Pull request closed without merging**